### PR TITLE
fixed : added attributes & latest version

### PR DESCRIPTION
**Pull request by pranaydeokar**
## what

- added attributes & latest version
- removed deprecated attributes and added attributes 
- updated and added .github workflows and readme.yaml file
- added outputs in examples 
- fixed tflint warning issues

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -255,29 +247,21 @@ resource "azurerm_monitor_diagnostic_setting" "acr-diag" {
   storage_account_id         = var.storage_account_id
   log_analytics_workspace_id = var.log_analytics_workspace_id
 
-  dynamic "log" {
-    for_each = var.acr_diag_logs
+  dynamic "enabled_log" {
+    for_each = var.log_enabled ? ["allLogs"] : []
```
Comment (13archit) [inline main.tf:251]: why directly we are passing `allLogs` can we not enable other types of logs?

**Review by 13archit — COMMENTED** [review]
Can we use meaningful for resources in module, not just pep1 or dnszone1

File: variables.tf [hunk 2]
```diff
@@ -54,6 +57,13 @@ variable "container_registry_config" {
   })
 }
 
+#azure_service_bypass
+variable "azure_services_bypass" {
+  description = "Whether to allow trusted Azure services to access a network restricted Container Registry? Possible values are None and AzureServices. Defaults to AzureServices"
```
Comment (13archit) [inline variables.tf:62]: follow a standard, description should be at last, first `type` than `default` than `descrption`

File: variables.tf [hunk 3]
```diff
@@ -54,6 +57,13 @@ variable "container_registry_config" {
...
+  type        = string
+  default     = "AzureServices"
+}
+
 variable "georeplications" {
   description = "A list of Azure locations where the container registry should be geo-replicated"
```
Comment (13archit) [inline variables.tf:68]: same for this, follow a standard.

File: _example/with_existing_dns_zone_in_diff_subs/output.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# output "resource_group_id" {
```
Comment (13archit) [inline _example/with_existing_dns_zone_in_diff_subs/output.tf:1]: why everything is commented?

File: _example/with_existing_dns_zone_in_diff_rg/output.tf [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# output "resource_group_id" {
```
Comment (13archit) [inline _example/with_existing_dns_zone_in_diff_rg/output.tf:1]: why everything is commented.

File: main.tf (hunk 1 above)
Comment (d4kverma) [reply main.tf:251]: this is a better option to get detailed logs.

File: variables.tf (hunk 2 above)
Comment (d4kverma) [reply variables.tf:62]: @pranaydeokar Please fix it.

File: _example/with_existing_dns_zone_in_diff_rg/output.tf (hunk 5 above)
Comment (d4kverma) [reply _example/with_existing_dns_zone_in_diff_rg/output.tf:1]: @pranaydeokar ?

File: main.tf (hunk 1 above)
Comment (13archit) [reply main.tf:251]: we can set it by default, as some might want only audit logs or something else.

**Pull request merged**